floris/tools/visualization.py

```diff
@@ -315,3 +319,139 @@ def plot_rotor_values(
+            y_bounds (tuple, optional): Limits of output array (in m).
+                Defaults to None.
+            wd (float, optional): Wind direction for visualization
+                Defaults to None.
+            ws (float, optional): Wind speed for visualization
+                Defaults to None.
+            yaw_angles ((np.ndarray), optional): Wind speed for visualization
+                Defaults to None.
+
+        Returns:
+            :py:class:`~.tools.cut_plane.CutPlane`: containing values
+            of x, y, u, v, w
+        """
+
+        # Make a local copy of fi to avoid editing passed in fi
+        fi = copy.deepcopy(fi_in)
+
+        # If wd/ws not provided, use what is set in fi
+        if wd is None:
+            wd = fi.floris.flow_field.wind_directions
+        if ws is None:
+            ws = fi.floris.flow_field.wind_speeds
+        fi.check_wind_condition_for_viz(wd=wd, ws=ws)
+
+        # # Store the current state for reinitialization
+        # floris_dict = fi.floris.as_dict()
+        # current_yaw_angles = fi.floris.farm.yaw_angles
+
+        # Set the ws and wd
+        fi.reinitialize(
+            wind_directions=wd, wind_speeds=ws
+        )
+
+        # Re-set yaw angles
+        if yaw_angles is not None:
+            fi.floris.farm.yaw_angles = yaw_angles
+
+        # Grab the turbine layout
+        layout_x = copy.deepcopy(fi.layout_x)
+        layout_y = copy.deepcopy(fi.layout_y)
+        D = fi.floris.farm.rotor_diameters_sorted[0][0][0]
+
+        # Declare a new layout array with an extra turbine
+        layout_x_test = np.append(layout_x,[0])
+        layout_y_test = np.append(layout_y,[0])
+
+
+        # Get a grid of points test test
+        if x_bounds is None:
+            x_bounds = (np.min(layout_x) - 2 * D, np.max(layout_x) + 10 * D)
+
+        if y_bounds is None:
+            y_bounds = (np.min(layout_y) - 2 * D, np.max(layout_y) + 2 * D)
+
+        # Now generate a list of points
+        x_points = np.linspace(x_bounds[0], x_bounds[1], x_resolution)
+        y_points = np.linspace(y_bounds[0], y_bounds[1], y_resolution)
+        num_points = len(x_points) * len(y_points)
+        
+        # Now loop over the points
+        x_results = np.zeros(num_points)
+        y_results = np.zeros(num_points)
+        z_results = np.zeros(num_points)
+        u_results = np.zeros(num_points)
+        v_results = np.zeros(num_points)
+        w_results = np.zeros(num_points)
+        idx = 0
+        for y in y_points:
+            for x in x_points:
+                
+                # Save the x and y results
+                x_results[idx] = x
+                y_results[idx] = y
+
+                # Place the test turbine at this location and calculate wake
+                layout_x_test[-1] = x
+                layout_y_test[-1] = y
+                fi.reinitialize(layout=(layout_x_test, layout_y_test))
+                fi.calculate_wake(yaw_angles=yaw_angles)
```

Comment:
I believe we need to down select the yaw angle needed here for the individual turbine.